crates/sui-core/src/authority.rs

```diff
@@ -1029,6 +1029,14 @@ impl AuthorityState {
         if !self.is_fullnode(&epoch_store) {
             return Err(anyhow!("dry-exec is only supported on fullnodes"));
         }
+        match transaction.kind() {
+            TransactionKind::ProgrammableTransaction(_) => (),
+            TransactionKind::ChangeEpoch(_)
+            | TransactionKind::Genesis(_)
+            | TransactionKind::ConsensusCommitPrologue(_) => {
+                return Err(anyhow!("dry-exec does not support system transactions"));
+            }
```

Comment:
And we know that these system txns can't be inside a ProgrammableTransaction?

Reply:
Yes

Reply:
There is `is_system_tx`